File: src/010/p010.py
import itertools
from src.utils.py.prime import generate_primes, primes


def solve1() -> int:
    return sum(itertools.takewhile(lambda num: num < 2000000, primes()))


def solve() -> int:
    # Prime sieve approach is much faster than using a generator
    return sum(generate_primes(2000000))

File: src/010/p010.go
package problem010

import (
	"strconv"

	"github.com/sunilbpandey/project-euler/src/utils/go/primeutils"
)

func Solve() string {
	primes := primeutils.GeneratePrimes(2000000)
	primeSum := 0
	for _, p := range primes {
		primeSum += p
	}
	return strconv.Itoa(primeSum)
}
